phac-nml/irida-next
# Dashboard groups

## What does this PR do and why?
Update the groups listing page to the dashboard groups page.

- Tree structure to how GitLab handles nested groups
- Stat area currently displaying how many subgroups are in a group
- Clicking on `>` or row will open subtree
- Clicking on name will take you to the group page

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Collapsed**
![image](https://github.com/phac-nml/irida-next/assets/11295750/33346fb0-8e7a-47f9-aa96-b6186eafd890)

**Open Group**
![image](https://github.com/phac-nml/irida-next/assets/11295750/11964ad3-e4e6-48d1-ac6f-dfeb1a6b4565)

**Dark Mode**
![image](https://github.com/phac-nml/irida-next/assets/11295750/315caa3b-5f37-45ae-a9c0-f40a6df4bed5)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

- Re-seed database
- Go to `/dashboard/groups`
- Click on the row (not the group link) for _Streptococcus_ to open the first level of the tree
- Click on the row for _Streptococcus pyogenes_ to see that subtree
- Try open and closing subtrees

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

  the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.
